src/modules/image-upload/ImageUploadHelper.js

```diff
@@ -63,17 +63,19 @@ export default class ImageUploadHelper {
 	}
 
 	_pollThumbnail(thumbnail: string): Promise<string> {
+		const TIMEOUT_INTERVAL = 15000;
+
 		return new Promise((resolve, reject) => {
 			const startTime = Date.now();
 
 			let thumbTimer;
 
 			const checkThumb = () => {
 				const onError = () => {
-					if (Date.now() - startTime > 15000) {
+					if (Date.now() - startTime > TIMEOUT_INTERVAL) {
 						reject(new Error('ERR_THUMBNAIL_TIMEOUT'));
 					} else {
-						thumbTimer = setTimeout(checkThumb, 1500);
```

Comment:
This does not seem correct. You check for the thumb every 1.5s, try 10 times before failing, correct? The new code checks only once, after 15 seconds.